analyzers/src/SonarAnalyzer.Common/SymbolicExecution/Roslyn/ExplodedNode.cs

```diff
@@ -46,8 +46,11 @@ private ExplodedNode(BasicBlock block, IOperationWrapperSonar[] operations, int
             programPointHash = ProgramPoint.Hash(block, index);
         }
 
-        public ExplodedNode CreateNext(ProgramState state) =>
-            new(Block, operations, index + 1, state, FinallyPoint);
+        public ExplodedNode CreateNext(ProgramState state)
+        {
+            state?.CheckStateConsistency();
```

Comment:
I don't understand why is this check here. Can you move it inside the constructor instead?

The null checking seems weird here. It appears to only solve the problem of reporting argument null inside consructor.